src/task_controller.cpp

```diff
@@ -327,9 +348,33 @@ bool MyTCServer::on_value_command(std::shared_ptr<isobus::ControlFunction> partn
 		{
 			std::uint8_t sectionIndexOffset = NUMBER_SECTIONS_PER_CONDENSED_MESSAGE * static_cast<std::uint8_t>(dataDescriptionIndex - static_cast<std::uint16_t>(isobus::DataDescriptionIndex::ActualCondensedWorkState1_16));
 
+			// Check if ActualWorkState is off (0) for either the current element or element 0 (main implement)
```

Comment:
Ultimately we should check this for complete hierarchy, so check recursively the parent element